system/security/shared/caching.hpp

```diff
@@ -151,16 +151,25 @@ class CPermissionsCache
 
     virtual ~CPermissionsCache();
 
-    static CPermissionsCache* queryInstance()
+    //Returns a shared cache of a given class type.
+    //Call this method with a unique class string ("LDAP", "MyOtherSecMgr")
+    //to create a cache shared amongst security managers of the same class
+    static CPermissionsCache* queryInstance(const char * _secMgrClass)
     {
+        const char * secMgrClass = (_secMgrClass != nullptr  &&  *_secMgrClass) ? _secMgrClass : "genericSecMgrClass";
+        typedef map<string, CPermissionsCache*> MapCache;
+        static MapCache m_mapCache;
+
+        CriticalBlock block(PCCritSect);
+        MapCache::iterator it = m_mapCache.find(secMgrClass);
+        if (it != m_mapCache.end())//exists in cache
+            return (*it).second;
+        else
         {
-            CriticalBlock block(PCCritSect);
-            if (instance == nullptr)
-            {
-                instance = new CPermissionsCache();
-            }
+            CPermissionsCache * instance = new CPermissionsCache();
+            m_mapCache.insert(pair<string, CPermissionsCache*>(secMgrClass, instance));
+            return instance;
         }
-        return instance;
```

Comment:
Instance was previously a static variable, I am guessing? Should it be deleted if no longer used?